test/unit/user_test.rb

```diff
@@ -74,10 +75,26 @@ class UserTest < ActiveSupport::TestCase
     test 'user model cannot be created without pin' do
       @user.pin = ""
       @user.save
-      assert_equal(["can't be blank", "may only contain numbers", "must be 4 digits."], @user.errors.messages[:pin])
+      assert_equal(["can't be blank", "is invalid", "must be 4 to 32 characters."], @user.errors.messages[:pin])
     end
   end
 
+
+  [generate_email].each do |new_email|
+    test 'validate pin length less than 4 characters' do
+      @user.pin = "123"
+      @user.save
+      assert_equal(["must be 4 to 32 characters."], @user.errors.messages[:pin])
+    end
+
+    test 'validate pin length more than 32 characters' do
+      @user.pin = "1234!qwertyuioplkjhgfsaqwerwuytrwesrs"
```

Comment:
Hi Deepika, sorry, this pin/password is 38 characters long! I would like to see one test for a 32 character-long password that _does_ pass validation, and a 33 character-long password that _doesn't_ pass validation. Thank you!